parser/openfoodfacts_taxonomy_parser/parser.py

```diff
@@ -15,11 +15,37 @@ def ellipsis(text, max=20):
     return text[:max] + ("..." if len(text) > max else "")
 
 
+class ParserConsoleLogger:
+    def __init__(self):
+        self.parsing_warnings = []  # Stores all warning logs
+        self.parsing_errors = []  # Stores all error logs
+
+        logging.basicConfig(
+            handlers=[logging.StreamHandler()],
+            level=logging.INFO,
+        )
```

Comment:
We must keep this out of here.
It's up to the caller to configure this, you really don't want this here.

You should move it back to the `if __name__ == "__main__"` part.

Reply:
Since this class is used for all the logging operations, I wanted to do the configuration when the ParserLogger class was created.

If we move it to the __main__ part, the config will only be done when the parser is run as a separate file. It will not be configured, when the `entries.py` file creates the Parser class.

Reply:
Yes but that's not the role of this class to setup logs for entries.py !

The FastAPI server should setup logs for itself and no sub part should change that.

That is, it's not the role of a sub-part of the code to setup loging, it's the role of the main program.

For the FastAPI server let setup log in api.py if you want, or in a settings.py file.

Reply:
I found a citation to explain it better: https://docs.python.org/3.8/howto/logging.html#configuring-logging

> It is strongly advised that you do not add any handlers other than [NullHandler](https://docs.python.org/3.8/library/logging.handlers.html#logging.NullHandler) to your library’s loggers. This is because the configuration of handlers is the prerogative of the application developer who uses your library. The application developer knows their target audience and what handlers are most appropriate for their application: if you add handlers ‘under the hood’, you might well interfere with their ability to carry out unit tests and deliver logs which suit their requirements.

Reply:
Yep, understood @alexgarel. Thanks for the help!